lua/packer/window.lua

```diff
@@ -0,0 +1,77 @@
+local window = {}
+
+local function default_win_opts()
+  local opts = {
+    relative = 'editor',
+    style    = 'minimal'
+  }
+
+  return opts
+end
+
+
+--- Create window that takes up certain percentags of the current screen.
+---
+--- Works regardless of current buffers, tabs, splits, etc.
+--@param col_range number | Table:
+--                  If number, then center the window taking up this percentage of the screen.
+--                  If table, first index should be start, second_index should be end
+--@param row_range number | Table:
+--                  If number, then center the window taking up this percentage of the screen.
+--                  If table, first index should be start, second_index should be end
+window.percentage_range_window = function(col_range, row_range, options)
```

Comment:
This seems like it might be better suited to live in `packer.display`, but that's a minor point and easily fixed up after merge.

Reply:
Yeah, just copied it in from another project of mine, so wanted to leave it separate for now. Can be merged into display whenever you want (I'll leave it to you to decide how)

Reply:
Sounds good; I'll leave it where it is at least until everything else is working. I want to do some refactoring of the overall codebase soon anyway (I think a lot of the git code, in particular, could be cleaned up and deduplicated), so I might handle all that after we merge this PR.